Editor/Window/Utils/StateManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿using AmazonGameLift.Editor;
+using AmazonGameLiftPlugin.Core.ApiGatewayManagement;
+
+namespace Editor.Window.Utils
+{
+    public class StateManager
+    {
+        private State _state;
+        public CoreApi CoreApi { get; set; }
+        public GameLiftRequestAdapter RequestAdapter { get; set; }
+        public State State { get; set; }
+        public IAmazonGameLiftClientWrapper GameLiftWrapper { get; private set; }
+        
+        public IAmazonGameLiftClientFactory AmazonGameLiftClientFactory { get; set; }
+        
+        public void SetupClientFactory()
+        {
+            AmazonGameLiftClientFactory = new AmazonGameLiftClientFactory(CoreApi);
```

Comment:
This should be part of construction since there's no reason we wouldn't have the CoreApi at start

Reply:
The problem with this is that we might use a different CoreApi that is constructed using a different path for settings to store UI information.

Reply:
Resolved in state management PR